src/filelocking.c

```diff
@@ -58,6 +59,21 @@ typedef struct _lockinfo {
 static unsigned int num_locked_files = 0;
 static lockinfo locked_files[MAX_LOCKED_FILES];
 
+int make_temp_file(char *tpl)
+{
+#if defined _MSC_VER || defined __MINGW32__
+    return _mktemp_s(tpl, strlen(tpl) + 1);
+#else
+    int _temp_fd = mkstemp(tpl);
+    if (!_temp_fd)
+        return EINVAL;
+    else {
+        close(_temp_fd);
+        return 0;
+    }
```

Comment:
Unnecessary else after a return:
```suggestion
    close(_temp_fd);
    return 0;
```